### ISSUE TITLE
amazon-ebssurrogate should send snapshot tags at snapshot creation time

### ISSUE THREAD

**Issue by nshalman** [hashicorp/packer#10144]
#### Community Note

Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request.
If you are interested in working on this issue or have submitted a pull request, please leave a comment.

#### Description

Some organizations configure their AWS accounts to require the presence of certain tags on everything.
In such an org, the code as it exists today will fail to take a snapshot of the ebs volume that the ebssurrogate has finished preparing. If Packer sends the tags when taking a snapshot, the snapshot will succeed.

Example code that worked for me to solve this problem (possibly appropriate for PR, but I'm not sure):
https://github.com/hashicorp/packer/compare/master...nshalman:nshalman/tag-snapshots

The only testing my example code has received is that it was able to build me an image where the current stable release could not.

**Comment by azr** [comment]
Hey @nshalman ! Thanks for opening.


>Example code that worked for me to solve this problem (possibly appropriate for PR, but I'm not sure):
master...nshalman:nshalman/tag-snapshots

 This code looks appropriate for PR _to me_. 🙂

**Comment by nshalman** [comment]
I have opened #10150 but I would like to note a few things:
1. The CLA feels heavy handed for code that is pretty much code copy-and-pasted from elsewhere in the code base. I did said copy-and-paste work during work hours with my work computer which means I have to do a bunch of due-diligence before I can comfortably complete it the CLA.
2. The PR template requests tests. I don't really know how to write the tests for this code, nor do I have free time to do so. I have solved my own problem, and while I think it would be wise for a comparable fix to be integrated into the code base, I have other priorities now that I need to get back to.

I kind of anticipated  both of those issues which is why I opened a feature request rather than a PR. Sadly I feel rather vindicated in that choice now. :(

Additionally, while my changes work, I'm not convinced that semantically they does what other users would expect. I think someone with more familiarity with the code base would do a better job writing a testable and correct implementation.
